schema-derivatives-options.graphql

```diff
@@ -0,0 +1,1302 @@
+
+ " Total number of contracts minted "
+  contractsMintedCount: Int!
+
+  " Total number of contracts taken "
+  contractsTakenCount: Int!
+
+ " Total number of contracts expired "
+  contractsExpiredCount: Int!
+
+  " Total number of contracts exercised "
+  contractsExercisedCount: Int!
+ 
+ " Total number of contracts exercised+expired "
+  contractsClosedCount: Int!
+
+  " Total number of open positions (minted-closed)"
+  openPositionCount: Int!
+
+  " Total number of closed positions (closed+exercised+expired) "
+  closedPositionCount: Int!
+
+  " Amount of input tokens in the pool. The ordering should be the same as the pool's `inputTokens` field. "
+  inputTokenBalances: [BigInt!]!
+
+  " Weights of input tokens in the liquidity pool in percentage values. For example, 50/50 for Uniswap pools, 48.2/51.8 for a Curve pool, 10/10/80 for a Balancer pool "
+  inputTokenWeights: [BigDecimal!]!
+
+  " Total supply of output token. Note that certain DEXes don't have an output token (e.g. Bancor) "
+  outputTokenSupply: BigInt
+
+  " Price per share of output token in USD "
+  outputTokenPriceUSD: BigDecimal
+
+  " Total supply of output tokens that are staked (usually in the MasterChef contract). Used to calculate reward APY. "
+  stakedOutputTokenAmount: BigInt
+
+  " Per-block reward token emission as of the current block normalized to a day, in token's native amount. This should be ideally calculated as the theoretical rate instead of the realized amount. "
+  rewardTokenEmissionsAmount: [BigInt!]
+
+  " Per-block reward token emission as of the current block normalized to a day, in USD value. This should be ideally calculated as the theoretical rate instead of the realized amount. "
+  rewardTokenEmissionsUSD: [BigDecimal!]
+
+  ##### Snapshots #####
+
+  " Liquidity pool daily snapshots "
+  dailySnapshots: [LiquidityPoolDailySnapshot!]! @derivedFrom(field: "pool")
+
+  " Liquidity pool hourly snapshots "
+  hourlySnapshots: [LiquidityPoolHourlySnapshot!]! @derivedFrom(field: "pool")
+
+  ##### Events #####
+
+  " All deposit (add liquidity) events occurred in this pool "
+  deposits: [Deposit!]! @derivedFrom(field: "pool")
+
+  " All withdraw (remove liquidity) events occurred in this pool "
+  withdraws: [Withdraw!]! @derivedFrom(field: "pool")
+
+  " All Position events occurred in this pool "
+  positions: [Position!]! @derivedFrom(field: "pool")
+}
+
+#################################
+##### Pool-Level Timeseries #####
+#################################
+
+type LiquidityPoolDailySnapshot @entity(immutable: true) {
+  " { Smart contract address of the pool }-{ # of days since Unix epoch time } "
+  id: Bytes!
+
+  " Number of days since Unix epoch time "
+  days: Int!
+
+  " Number of days since Unix epoch time "
+  days: Int!
+
+  " The protocol this snapshot belongs to "
+  protocol: DerivOptProtocol!
+
+  " The pool this snapshot belongs to "
+  pool: LiquidityPool!
+
+  " Block number of this snapshot "
+  blockNumber: BigInt!
```

Comment:
@jaimehgb have we made a decision on keeping or removing `blocknum` / `timestamp`. I thought that if the snapshot was immutable then it would not be possible to get a proper block number/timestamp within that snapshot, so we were just going to have days/hours

Reply:
we removed `blocknumber` and `timestamp` from the immutable time series. thanks for catching this!